pkg/internal/cluster/create/actions/config/config.go

```diff
@@ -158,7 +181,7 @@ func allPatchesFromConfig(cfg *config.Cluster) (patches []string, jsonPatches []
 }
 
 // writeKubeadmConfig writes the kubeadm configuration in the specified node
-func writeKubeadmConfig(logger log.Logger, cfg *config.Cluster, data kubeadm.ConfigData, node nodes.Node) error {
+func writeKubeadmConfig(cfg *config.Cluster, kubeadmConfig string, data kubeadm.ConfigData, node nodes.Node) error {
```

Comment:
this is wrong, we should not be taking in data and the kubeadm config.
we fix up the data below, but we've already generated the config

Reply:
Good catch! It turns out that there were some **"data"** configs that I missed. Fixing...